src/utils/transaction.ts

```diff
@@ -20,35 +22,121 @@ import {
   getDecimalFromEtherUnit,
   gasPriceToBase
 } from '@services/EthService';
-import { ITxReceipt, ExtendedAsset, Network, ITxConfig, StoreAccount, TAddress } from '@types';
+import {
+  ExtendedAsset,
+  Network,
+  ITxConfig,
+  StoreAccount,
+  TAddress,
+  ITxType,
+  ITxStatus,
+  IPendingTxReceipt,
+  ITxHash,
+  IFailedTxReceipt,
+  ISuccessfulTxReceipt,
+  ITxReceipt,
+  ITxHistoryStatus
+} from '@types';
+
+export const constructTxReceiptFromTransactionResponse = (txResponse: TransactionResponse) => (
+  txType: ITxType,
+  txConfig: ITxConfig,
+  assets: ExtendedAsset[],
+  txHash?: ITxHash
+) => (status: ITxStatus.PENDING): IPendingTxReceipt => {
+  const useTxResponse = txResponse && !isEmpty(txResponse) && txResponse !== null;
```

Comment:
`txResponse` is `null` or undefined if the tx is not found on a node yet. This can occur if we send the tx to one node (as an example: using metamask, it goes through an infura node), and then we attempt to fetch the txResponse from a different node (maybe mycrypto's node cluster, or a different node in infura's node cluste) before it has received the tx in it's mempool. The actual return seems to depend on the node or time it's hitting. Sometimes it'll just be an empty return, sometimes it returns null. Not sure why though.

We don't want to fail out when this happens, because the tx would have to have been broadcast correctly before we got here, so we want to construct the tx mostly from the tx config.

Reply:
Let's use the hash returned by the wallets and use txConfig for the rest.